src/extension-runners/chromium.js

```diff
@@ -232,6 +233,7 @@ export class ChromiumExtensionRunner {
       userDataDir,
       // Ignore default flags to keep the extension enabled.
       ignoreDefaultFlags: true,
+      port: this.params.chromiumDebuggerPort,
```

Comment:
Passing this flag without validation is dangerous, because it may break the functionality of `web-ext`.

`web-ext` has to launch a new Chrome browser instance, because the extension and auxiliary extension are installed via a command-line flag, `--load-extension` to be specific.

When `chrome-launcher` receives a given port, it tries to connect to that port, and if it exists, it will not start a new Chrome browser, but connect to an existing one instead: https://github.com/GoogleChrome/chrome-launcher/blob/v0.15.1/src/chrome-launcher.ts#L248-L264

IF we want to support taking a fixed flag as input, then we should validate that the port is unused, and throw an explicit error if it is used.

Reply:
I agree with the basic idea that not validating external parameters is dangerous, but I don't think it applies in this case. web-ext is just passing a port number, and it is essentially the browser's responsibility to validate the port number. Well, chrome doesn't seem to check it.

Even if you implement the verification code, wouldn't it be enough to verify that the specified number is within the TCP port range?

Verifying whether a port is in use may be useful, but it is a "nice to have" feature.
Considering that the user of web-ext is a developer, it is common sense not to use duplicate ports, and even if duplicate ports are used by mistake, they will immediately consider the possibility.

So the bottom line is that for most web developers, the ability to verify that a port is in use is an unnecessary feature. This is undesirable because it adds complexity to the code and reduces maintainability.

Reply:
I explained why the port has to be unused in my previous comment.

Summarized, web-ext would not work and be unreliable. The `web-ext` program expects the chrome-launcher library to launch the browser with the extension loaded, but if the port is in use, it won't launch Chrome, and therefore the core feature of `web-ext run` will be stuck waiting for the browser+extension to start. Explicitly failing with a useful message is more desirable than getting stuck without further information.

Reply:
When I get time to work on it, I will check the stacking behavior.